firmware/examples/fdt-read/memory.x

```diff
@@ -6,12 +6,13 @@ SPDX-License-Identifier: CC0-1.0
 
 MEMORY
 {
-  RAM : ORIGIN = 0x80000000, LENGTH = 256K
+  DATA : ORIGIN = 0x40000000, LENGTH = 64K
+  INSTR : ORIGIN = 0x20000000, LENGTH = 64K
 }
 
-REGION_ALIAS("REGION_TEXT", RAM);
-REGION_ALIAS("REGION_RODATA", RAM);
-REGION_ALIAS("REGION_DATA", RAM);
-REGION_ALIAS("REGION_BSS", RAM);
-REGION_ALIAS("REGION_HEAP", RAM);
-REGION_ALIAS("REGION_STACK", RAM);
+REGION_ALIAS("REGION_TEXT", INSTR);
```

Comment:
I'm not particularly well-known with the embedded stuff, but why would `REGION_TEXT` live in the `INSTR` memory space, while the other ones live in `DATA`?

Reply:
"text" is old terminology for "instructions", if you have an ELF file, the actual instructions will be in a `.text` section.

This separation here is to have some kind of Harvard architecture where the instruction and data buses are separate. All other things will be accessed via the data bus, so it's in the data memory, while instructions are accessed through the instruction bus.

I think eventually we want to merge those two and have a von-Neumann architecture, but all that's needed here would be to change it back to one unified RAM